sdk/test-utils/recorder/README.md

```diff
@@ -1,18 +1,516 @@
+
+This library provides interfaces and helper methods to equip the SDKs in the `azure-sdk-for-js` repo with the recording and playback capabilities for the tests, it targets HTTP requests in both Node.js and the Browsers.
+
+`@azure-tools/test-recorder`, as part of the Test Utils available in this repository, it is supposed to be added only as a devDependency and should be used only for the tests of an sdk.
 
 This tool helps to record and playback the tests in the JS repo by leveraging the unified out-of-process test proxy server.
 
-## Resources
+## Index
+
+- [Key concepts](#key-concepts)
+- [Getting started](#getting-started)
+  - [Installing the package](#installing-the-package)
+  - [Configuring your project](#configuring-your-project)
+  - [Using the `Recorder`](#using-the-recorder)
+- [Examples](#examples)
+  - [How to record](#how-to-record)
+  - [How to playback](#how-to-playback)
+  - [Update existing recordings](#update-existing-recordings)
+  - [Skipping tests](#skipping-tests)
+  - [Securing sensitive data](#securing-sensitive-data)
+  - [Supporting parallelism](#supporting-parallelism)
+  - [Isomorphic tests](#isomorphic-tests)
+  - [Many ways to run the test-proxy tool](#many-ways-to-run-the-test-proxy-tool)
+- [Troubleshooting](#troubleshooting)
+- [Next steps](#next-steps)
+- [Contributing](#contributing)
+
+## Key concepts
+
+- To **record** means to intercept any HTTP request, store it in a file, then store the response received from the live resource that was originally targeted. We leverage the unified out-of-process test proxy server that is built for this use case. The output files are stored in `recordings/node/*` and in `recordings/browser/*`, which are relative to the root of the project you're working on.
+- To **playback** means to intercept any HTTP request and to respond it with the stored response of a previously recorded matching request.
+- **Sensitive information** means content that should not be shared publicly. Content like passwords, unique identifiers or personal information should be cleaned up from the recordings. Some functionality is provided to fix this problem. You can read more at [securing sensitive data](#securing-sensitive-data).
+
+## Getting started
+
+We're about to go through how to set up your project to use the `@azure-tools/test-recorder` package.
+
+This document assumes familiarity with [git](https://git-scm.com) and [rush](https://rushjs.io). You can read more about how we use rush in the following links:
+
+- Rush used for [Project Orchestration](https://github.com/sadasant/azure-sdk-for-js/blob/master/CONTRIBUTING.md#project-orchestration).
+- [Rush for NPM users](https://github.com/sadasant/azure-sdk-for-js/blob/master/CONTRIBUTING.md#rush-for-npm-users).
+
+### Installing the package
```

Comment:
I am confused by this section because it explains how to close and build the repo, things I would expect to be in the contributing guide but not in an internal package readme. Could we instead just put a command to add the package to package.json, e.g. `rush add -p @azure-tools/test-recorder@^2.0.0 --dev`? the document can then refer to the contributing guide for instructions on how to link and build the deps.

Reply:
Great point, I'll update this.
Retained this from recorder v1, but I would agree with what you said.

Reply:
Updated some things, is it nicer now?